### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by RomSmile**
DEMO LINK: https://RomSmile.github.io/react_person/

### CODE REVIEW

File: src/components/InfoAboutPerson.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+function InfoAboutPerson({ person }) {
+  return (
+    <section className="Person">
+      <h2 className="Person__name">
+        {person.name ? `My name is ${person.name}` : null}
+      </h2>
+
+      <p className="Person__age">
+        I am
+        {person.age || '0'}
```
Comment (AndreySerhovetc) [inline src/components/InfoAboutPerson.jsx:13]: person.age || null

**Review by AndreySerhovetc — APPROVED** [review]

File: src/App.js [hunk 2]
```diff
@@ -25,22 +26,9 @@ const alex = {
...
-      <h2 className="Person__name">My name is Olya</h2>
-      <p className="Person__partner">My husband&apos;s name is Maksym</p>
-    </section>
-
-    <section className="Person">
-      <h2 className="Person__name">My name is Alex</h2>
-      <p className="Person__age">I am 25</p>
-      <p className="Person__partner">I am not married</p>
-    </section>
+    <InfoAboutPerson person={misha} />
```
Comment (varseniuk) [inline src/App.js:29]:
```suggestion
    <Person person={misha} />
```
Read task description more carefully:

> Create a Person component

File: src/components/InfoAboutPerson.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+InfoAboutPerson.propTypes = {
+  person: PropTypes.shape({
+    name: PropTypes.string.isRequired,
+    age: PropTypes.number,
+    sex: PropTypes.string.isRequired,
+    isMarried: PropTypes.bool.isRequired,
+    partnerName: PropTypes.string,
+  }).isRequired,
```
Comment (varseniuk) [inline src/components/InfoAboutPerson.jsx:32]: Add default props for `age` and `partnerName`

File: src/components/InfoAboutPerson.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function InfoAboutPerson({ person }) {
+  return (
+    <section className="Person">
+      <h2 className="Person__name">
+        {person.name ? `My name is ${person.name}` : null}
+      </h2>
```
Comment (varseniuk) [inline src/components/InfoAboutPerson.jsx:9]: `name` is required. It cannot be null
```suggestion
      <h2 className="Person__name">
        My name is {person.name}
      </h2>
```

File: src/components/InfoAboutPerson.jsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+      <p className="Person__age">
+        I am
+        {person.age || '0'}
+      </p>
```
Comment (varseniuk) [inline src/components/InfoAboutPerson.jsx:14]:
```suggestion
      {age && (
        <p className="Person__age">
          I am {person.age}
        </p>
      )}
```
Use conditional rendering.

File: src/components/InfoAboutPerson.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      <p className="Person__partner">
+        {person.isMarried
+          ? `My wife's name is ${person.partnerName}`
```
Comment (varseniuk) [inline src/components/InfoAboutPerson.jsx:17]: partner have sex. Why they are always `wives` in your code?
Wife's name Maxym looks strange

Снимок экрана 2021-09-02 в 10 33 56

**Review by varseniuk — CHANGES_REQUESTED** [review]
Let's fix it.
Wife Maxym and age 0 looks very strange

Снимок экрана 2021-09-02 в 10 33 56

File: src/components/Person.jsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        {person.name}
+      </h2>
+
+      {person.age && (
+      <p className="Person__age">
+        I am
+        {' '}
+        {person.age}
+      </p>
+      )}
```
Comment (varseniuk) [inline src/components/Person.jsx:21]:
```suggestion
      {person.age && (
        <p className="Person__age">
          I am
          {' '}
          {person.age}
        </p>
      )}
```
Pay attention to the formatting

File: src/components/Person.jsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+      <p className="Person__partner">
+        {person.isMarried
+          ? `My ${wifeOrHusband} name is ${person.partnerName}`
+          : null
+        }
+      </p>
```
Comment (varseniuk) [inline src/components/Person.jsx:28]:
```suggestion
      {person.isMarried && (
        <p className="Person__partner">
          `My ${wifeOrHusband} name is ${person.partnerName}`
        </p>
      )}
```
You could use conditional formatting here as well

**Review by varseniuk — APPROVED** [review]
Good! Left some comments

**Pull request closed without merging**